cpp/linear_regression/avocado_price_prediction/avocado_price_prediction.cpp

```diff
@@ -0,0 +1,140 @@
+* Type - Conventional or organic.
+* Year - Year of observation.
+* Region - City or region of observation.
+*
+* Approach
+* 
+* In this example, first we will do EDA on the dataset to find correlation
+* between various features.
+* Then we'll be using onehot encoding to encode categorical features.
+* Finally we will use LinearRegression API from mlpack to learn the correlation
+* between various features and the target i.e AveragePrice.
+* After training the model, we will use it to do some predictions, followed by
+* various evaluation metrics to quantify how well our model behaves.
+*/
+#include <mlpack.hpp>
+using namespace mlpack;
+using namespace mlpack::data;
+
+//Drop the dataset header using sed, sed is a Unix utility that parses and transforms text."
+//!mkdir -p data && cat avocado.csv | sed 1d > avocado_trim.csv"
+//"Drop columns 1 and 2 (\"Unamed: 0\", \"Date\") as these are not required and their presence cause issues while loading the data."
+//!rm avocado_trim.csv"
+//"!mv avocado_trim2.csv avocado_trim.csv"
+
+int main()
+{
+ //!wget -q https://datasets.mlpack.org/avocado.csv.gz"
+ // Features 9 (Avocado type) and 11 (region of observation) are strings
+ // (categorical), but armadillo matrices can contain only numeric information;
+ // so, we have to explicitly define them as categorical in `datasetInfo`
+ // this allows mlpack to map numeric values to each of those values,
+ // which can later be unmapped to strings.
+ // Load the dataset into armadillo matrix.
+
+ arma::mat matrix;
+ data::DatasetInfo info;
+ info.Type(9) = data::Datatype::categorical;
+ info.Type(11) = data::Datatype::categorical;
+ data::Load("../../../data/avocado.csv", matrix, info);
+ // Printing header for dataset.
+ std::cout << std::setw(10) << "AveragePrice" << std::setw(14) 
+     << "Total Volume" << std::setw(9) << "4046" << std::setw(13) 
+     << "4225" << std::setw(13) << "4770" << std::setw(17) << "Total Bags" 
+     << std::setw(13) << "Small Bags" << std::setw(13) << "Large Bags" 
+     << std::setw(17) << "XLarge Bags" << std::setw(10) << "Type" 
+     << std::setw(10) << "Year" << std::setw(15) << "Region" <<  std::endl;
+ std::cout << matrix.submat(0, 0, matrix.n_rows-1, 5).t() << std::endl;
+ // Exploratory Data Analysis
+ arma::mat output;
+ data::OneHotEncoding(matrix, output, info);
+ arma::Row<double> targets = arma::conv_to<arma::Row<double>>::from(output.row(0));
+ // Labels are dropped from the originally loaded data to be used as features.
+ output.shed_row(0);
+ // Train Test Split,
+ // The dataset has to be split into a training set and a test set. Here the
+ // dataset has 18249 observations and the `testRatio` is set to 20% of the
+ // total observations. This indicates the test set should have
+ // 20% * 18249 = 3649 observations and training test should have
+ // 14600 observations respectively.
+ arma::mat Xtrain;
+ arma::mat Xtest;
+ arma::Row<double> Ytrain;
+ arma::Row<double> Ytest;
```

Comment:
```suggestion
 arma::rowvec Ytrain;
 arma::rowvec Ytest;
```
Probably better to use the more common name here.